src/modules/payment/forms/billing-adress-form.tsx

```diff
@@ -0,0 +1,120 @@
+import type { SubmitErrorHandler } from 'react-hook-form';
+import { z } from 'zod';
+
+import { withFormHOC } from '../../../core/hoc/with-form-hoc';
+import { common } from '../../../translations/en.json';
+import { ControlledInput } from '../../../ui/core/input';
+import { Text } from '../../../ui/core/text';
+import { View } from '../../../ui/core/view';
+
+const _billingSchema = z.object({
+  city: z.string({
+    required_error: 'City is required',
+  }),
+  country: z.string({
+    required_error: 'Country is required.',
+  }),
+  adress1: z.string({
+    required_error: 'First adress is required',
+  }),
+  adress2: z
+    .string({
+      required_error: 'Second adress is required',
+    })
+    .optional(),
+  postalCode: z.string({
+    required_error: 'Postal code is required',
+  }),
+});
+
+const schema = _billingSchema;
+
+export type TBillingAdressFormFields = z.infer<typeof schema>;
+
+export type BillingAdressFormSubmitHandler = (fields: TBillingAdressFormFields) => void;
+export type BillingAdressFormSubmitErrorHandler = SubmitErrorHandler<TBillingAdressFormFields>;
+
+const BillingAdressForm = withFormHOC(
+  {
+    schema,
+    config: {
+      shouldFocusError: false,
+      mode: 'onSubmit',
+    },
+  },
+  ({ formMethods, disabled }) => {
+    const { control } = formMethods;
+
+    return (
+      <>
+        <View className="flex-1">
+          <Text variant="h6-bold" className="mb-4 mt-7">
+            {common.labels.billingAdresss}
+          </Text>
+          <Text variant="h6" className="mb-1.5 mt-3">
+            {common.labels.city}
+            {common.labels.starSign}
+          </Text>
+          <ControlledInput
+            control={control}
+            name="city"
+            placeholder={common.labels.city}
+            editable={!disabled}
+          />
+        </View>
+        <View className="flex-1">
```

Comment:
This structure, <view><text/><controlledinput/></view> seems to be repeated for all the fields of the form, maybe you could create a FormField component and reuse it. Adding the starsign or not could depend on a boolean property called `mandatory`. 

You could also have an array of your fields and instead of having one block of code for each field you could have a fields constant and do something like 
```
fields.map(({label, name, placeholder, mandatory})=> 
<View className="flex-1">
          <Text variant="h6" className="mb-1.5 mt-3">
            {label}
            {mandatory ? common.labels.starSign : ''}
          </Text>
          <ControlledInput
            control={control}
            name={name}
            placeholder={placeholder}
            editable={!disabled}
          />
        </View>)
```

Reply:
Done